components/kserve/kserve.go

```diff
@@ -54,6 +54,8 @@ type Kserve struct {
 	// This field is optional. If no default deployment mode is specified, Kserve will use Serverless mode.
 	// +kubebuilder:validation:Enum=Serverless;RawDeployment
 	DefaultDeploymentMode DefaultDeploymentMode `json:"defaultDeploymentMode,omitempty"`
+	// Configures and enables Nvidida's NIM integration
+	NIM components.Component `json:"nim,omitempty"`
```

Comment:
This should not be a full component, you probably need a `NIMSpec` or something like that, similar to the `Serving` field above

Reply:
@lburgazzoli Can you help me understand what is your wanting here?
Are intent is simply have nim as sub component kserve with options of managed or removed which will trigger the necessary logic from odh-model-controller

Reply:
A [component](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/components/component.go) is quite a complex resource and for this case, I think we don't need a full component (in that case, you should be a top level component). 

In should be enough to add a dedicate spec here with the  required configuration options, and handle the deployment as part as the [Kserve reconciliation logic](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/components/kserve/kserve.go#L113).  Usually, this can be done by providing a set of kustomize manifests and conditionally deploy them.

Reply:
@lburgazzoli added nim spec. Please review and advise if other changes are needed.